cmd/axiom-syslog-proxy/main.go

```diff
@@ -38,7 +40,15 @@ func run(_ context.Context, _ *zap.Logger, client *axiom.Client) error {
 		return cmd.Error("create server", err)
 	}
 
-	srv.Run()
+	// Setup cancellation context that will be cancelled on receiving a signal
+	ctx, stop := signal.NotifyContext(ctx, syscall.SIGINT, syscall.SIGTERM)
+	defer stop()
 
-	return nil
+	go func() {
+		srv.Run()
+	}()
+
+	<-ctx.Done()
+
+	return ctx.Err()
```

Comment:
I don't think this change is needed. `github.com/axiomhq/pkg/cmd` deals with creating a proper cancellation context. Did you experience issues killing the binary?

Reply:
Yeah couldn't kill it and couldn't kill the docker one either

Reply:
Either pass the given context to `srv` or just run it in a go routine and wait for it, like you're doing at the moment but just don't create a new `NotifyContext`. The first approach would be way cleaner, tho :)

There is a `Shutdown(ctx)` method on Go's http server + some additional tricks. AFAIK, the `github.com/axiomhq/pkg` might have a speciality http server package, if you wanna go that route ;) 

But that's extra credit, not mandatory :D Just don't create a second `NotifyContext`.